File: 프로그래머스/2/132265. 롤케이크 자르기/롤케이크 자르기.js
function solution(topping) {
    var answer = 0;
    const map = new Map();
    const set = new Set();
    for (const t of topping) {
        map.set(t, (map.get(t) || 0) + 1);
    }
    for (const t of topping) {
        set.add(t);
        map.set(t, map.get(t) - 1);
        if (map.get(t) === 0) map.delete(t);
        
        if (set.size === map.size) answer++;
    }
    return answer;
}

File: 프로그래머스/2/132265. 롤케이크 자르기/롤케이크 자르기.java
import java.util.*;
class Solution {
    public int solution(int[] topping) {
        int answer = 0;
        HashMap<Integer, Integer> map1 = new HashMap<>();
        HashMap<Integer, Integer> map2 = new HashMap<>();
        
        for(int i : topping) {
            map2.put(i, map2.getOrDefault(i,0)+1);
        }
        for(int i : topping) {
            map1.put(i, map1.getOrDefault(i,0)+1);
            map2.put(i, map2.get(i)-1);
            if(map2.get(i) == 0) {
                map2.remove(i);
            }
            if(map1.size() == map2.size()) {
                answer++;
            }
        }
        
        return answer;
    }
}